### PR TITLE
tests

### PR DESCRIPTION
**Pull request by jaworskak**

### CODE REVIEW

File: backend/test/test_no_waste.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    assert recipe.serialize == {
+            "title": "Kielbasa",
+            "imgUrl": 'https://loremflickr.com/320/240/food',
+            "description": "Przepis na kielbase",
+            "ingredients": ["Kielbasa", "Chleb", "Ketchup"]
+        }
+
+
+def test_product(client):
+    request = 'm'
```
Comment (jakule) [inline backend/test/test_no_waste.py:31]: This is not being used. :confused:

File: backend/test_requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+pytest==6.2.1
```
Comment (jakule) [inline backend/test_requirements.txt:1]: By convention this file should be called: `dev-requirements.txt`
This file should also include `-r requirments.txt` at the beginning.

File: backend/test/test_no_waste.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from no_waste import no_waste
+from no_waste.no_waste import Recipe, Product
+
+
+import pytest
+
+@pytest.fixture
+def client():
```
Comment (jakule) [inline backend/test/test_no_waste.py:8]: This function is not used anywhere

File: backend/test/test_no_waste.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    return no_waste.app.test_client()
+
+
+def test_nonempty_initial_db():
+    assert Recipe.query.count() != 0
```
Comment (jakule) [inline backend/test/test_no_waste.py:13]: Each test should start with a clean state.

**Review by jakule — CHANGES_REQUESTED** [review]
This is what I'm getting when I try to run the tests:
```
% pytest .
====================================================== test session starts ======================================================
platform linux -- Python 3.8.6, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/jnyckowski/projects/no-waste/backend
collected 0 items / 1 error                                                                                                     

============================================================ ERRORS =============================================================
____________________________________________ ERROR collecting test/test_no_waste.py _____________________________________________
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1276: in _execute_context
    self.dialect.do_execute(
venv/lib/python3.8/site-packages/sqlalchemy/engine/default.py:609: in do_execute
    cursor.execute(statement, parameters)
E   sqlite3.OperationalError: no such column: recipes.ingredients

The above exception was the direct cause of the following exception:
test/test_no_waste.py:1: in <module>
    from no_waste import no_waste
no_waste/no_waste.py:49: in <module>
    if Recipe.query.count() == 0:
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3803: in count
    return self.from_self(col).scalar()
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3523: in scalar
    ret = self.one()
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3490: in one
    ret = self.one_or_none()
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3459: in one_or_none
    ret = list(self)
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3535: in __iter__
    return self._execute_and_instances(context)
venv/lib/python3.8/site-packages/sqlalchemy/orm/query.py:3560: in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1011: in execute
    return meth(self, multiparams, params)
venv/lib/python3.8/site-packages/sqlalchemy/sql/elements.py:298: in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1124: in _execute_clauseelement
    ret = self._execute_context(
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1316: in _execute_context
    self._handle_dbapi_exception(
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1510: in _handle_dbapi_exception
    util.raise_(
venv/lib/python3.8/site-packages/sqlalchemy/util/compat.py:182: in raise_
    raise exception
venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1276: in _execute_context
    self.dialect.do_execute(
venv/lib/python3.8/site-packages/sqlalchemy/engine/default.py:609: in do_execute
    cursor.execute(statement, parameters)
E   sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: recipes.ingredients
E   [SQL: SELECT count(*) AS count_1 
E   FROM (SELECT recipes.id AS recipes_id, recipes.recipe_name AS recipes_recipe_name, recipes.recipe AS recipes_recipe, recipes.ingredients AS recipes_ingredients 
E   FROM recipes) AS anon_1]
E   (Background on this error at: http://sqlalche.me/e/13/e3q8)
==================================================== short test summary info ====================================================
ERROR test/test_no_waste.py - sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: recipes.ingredients
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================= 1 error in 0.43s ========================================================
```

File: backend/test/test_no_waste.py (hunk 3 above)
Comment (jaworskak) [reply backend/test/test_no_waste.py:8]: It's injected as parameter to test_product

File: backend/test/test_no_waste.py (hunk 4 above)
Comment (jaworskak) [reply backend/test/test_no_waste.py:13]: There is an initial database state and this is what this test tests

**Comment by jaworskak** [comment]
there will be new tests with mongo

**Pull request closed without merging**